components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -99,8 +99,10 @@ public boolean doPreAddUserWithID(String userID, Object credential, String[] rol
             if (!isEnable() || userStoreManager == null || !userStoreManager.isSCIMEnabled()) {
                 return true;
             }
-            // Validate claim value against the regex.
-            validateClaimValue(claims, userStoreManager);
+            // Validate claim value against the regex if user claim input regex validation configuration is enabled.
+            if (SCIMCommonUtils.isRegexValidationForUserClaimEnabled()) {
```

Comment:
Have we enabled this configuration OOTB?

Reply:
Yes, default value has been set to `true` in [https://github.com/lashinijay/carbon-identity-framework/blob/431103c965c74425ddc2072304a0edd421d4f774/features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json#L369](https://github.com/lashinijay/carbon-identity-framework/blob/431103c965c74425ddc2072304a0edd421d4f774/features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json#L369)